src/WebJobs.Script/Host/DefaultFunctionMetadataProvider.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    internal class DefaultFunctionMetadataProvider : IFunctionMetadataProvider
```

Comment:
What do we mean by `default` in the naming here? This class is doing same thing as `AggregateFunctionMetadataProvider.cs`  class we had earlier which meant to working with both objects - `IWorkerFunctionMetadataProvider` and `IHostFunctionMetadataProvider` same as this class is doing. But, I feel _Aggregate_ makes more sense than _default_ to refer to combination of both providers.

We introduced the term `UseDefaultMetadataIndexing` to refer to Host indexing as default.

Reply:
Aggregation generally means we are combining more than one thing. However, the AggregateFunctionMetadataProvider wasn't a true aggregator. It was instead a selector. For something to be an aggregator it needs to combine results from multiple sources into one. I believe we have aggregateException that behaves in a similar manner. I do get your point though default in this case can be misleading. We can probably name it `FunctionMetadataProvider`